lib/class-wp-rest-post-format-search-handler.php

```diff
@@ -0,0 +1,117 @@
+<?php
+/**
+ * REST API: WP_REST_Post_Format_Search_Handler class
+ *
+ * @package Gutenberg
+ */
+
+/**
+ * Core class representing a search handler for post formats in the REST API.
+ *
+ * @see WP_REST_Search_Handler
+ */
+class WP_REST_Post_Format_Search_Handler extends WP_REST_Search_Handler {
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->type = 'post-format';
+	}
+
+	/**
+	 * Searches the object type content for a given search request.
+	 *
+	 * @param WP_REST_Request $request Full REST request.
+	 * @return array Associative array containing an `WP_REST_Search_Handler::RESULT_IDS` containing
+	 *               an array of found IDs and `WP_REST_Search_Handler::RESULT_TOTAL` containing the
+	 *               total count for the matching search results.
+	 */
+	public function search_items( WP_REST_Request $request ) {
+		$format_strings = get_post_format_strings();
```

Comment:
I'm not sure. What do you think? Right now it will only return post formats that have posts because of the call to `get_post_format_link()` below.

Reply:
I'm not sure either. I think it could go either way. The closest example would be the current nav menu screen which seems to show used formats even if the theme doesn't support them.

Reply:
Let's try it like this and see if we run into issues. We ostensibly have until 5.6 RC 1 (November 17) to make breaking changes.